### Задание 2.
Напишите Python-скрипт, который принимает путь к большому CSV-файлу и подсчитывает количество строк (записей) в файле без загрузки его целиком в память. Скрипт должен быть оптимизирован для работы с большими файлами.

**Решение 1.** С помощью простого построчного чтения файла

In [4]:
count = 0
with open('input.csv', 'r', encoding='utf-8') as file:
    for line in file:
        count += 1
        
print(count)

375142



**Решение 2.** С помощью библиотеки csv и обработчика csv файлов - csv.reader

In [ ]:
import csv

with open('input.csv', 'r', encoding='utf-8') as file:
    reader = csv.reader(file)
    count = sum(1 for row in reader)  # Подсчет строк
    print(count)

375142


**Первое решение** выполняется быстрее **второго**: `0.4s` против `2.9s` для моего файла размером 375142 строк. Это логично, т.к. функция `csv.reader` дополнительно обрабатывает csv файл, разделяет по запятым и пр. Поэтому оставим **решение 1**.

P.s. Если по заданию требовалось бы загружать весь файл целиком и при этом оптимизировать эту загрузку для больших файлов, можно было бы:
- использовать загрузку по частям (`chunks`) в `pandas`;
- использовать библиотеку `dask` для распределенных вычислений и параллельной обработки данных на нескольких ядрах или серверах.